### PR TITLE
use temp file for gcb_env.sh

### PR DESCRIPTION
**Pull request by rkpagadala**
airflow runs seem not to be fully isolated, use temp file for gcb_env.sh to eliminate conflicts

### CODE REVIEW

File: release/pipeline/airflow_scripts.sh [hunk 1]
```diff
@@ -32,59 +32,53 @@ cat << EOF > "${SUBS_FILE}"
     "_CB_GCS_RELEASE_TOOLS_PATH": "${CB_GCS_RELEASE_TOOLS_PATH}"
   }
 EOF
+   cat "${SUBS_FILE}"
```
Comment (hklai) [inline release/pipeline/airflow_scripts.sh]: why is this needed?

File: release/pipeline/airflow_scripts.sh [hunk 2]
```diff
@@ -23,7 +23,7 @@ SCRIPTPATH=$( pwd -P )
 # shellcheck source=release/pipeline/gcb_build_lib.sh
 source "${SCRIPTPATH}/gcb_build_lib.sh"
 
-# Helper function not called by Airflow, used only by functions in this file
+# Helper function called by Airflow before calling the actual work functions below
 function create_subs_file() {
```
Comment (hklai) [inline release/pipeline/airflow_scripts.sh]: if this function is always called, you may as well remove the function and put the few lines in top level under the above source "..."

File: release/pipeline/dags/istio_common_dag.py [hunk 3]
```diff
@@ -97,22 +102,25 @@ def getBashSettingsTemplate(extra_param_lst=[]):
...
 
   template_list.append("""
-                source    "/tmp/gcb_env.sh"
                 # download gcb_env.sh if it exists, otherwise the local copy will be uploaded
-                gsutil -q cp "gs://${CB_GCS_RELEASE_TOOLS_PATH}/gcb_env.sh" "/tmp/gcb_env.sh"
-                source    "/tmp/gcb_env.sh"
-                gsutil -q cp "/tmp/gcb_env.sh" "gs://${CB_GCS_RELEASE_TOOLS_PATH}/"
+                gsutil -q cp "gs://${CB_GCS_RELEASE_TOOLS_PATH}/gcb_env.sh" "${GCB_ENV_FILE}"
+                if [ $? -ne 0 ]; then
+                  gsutil -q cp "${GCB_ENV_FILE}" "gs://${CB_GCS_RELEASE_TOOLS_PATH}/gcb_env.sh"
```
Comment (hklai) [inline release/pipeline/dags/istio_common_dag.py]: not sure i follow.
why copy the file back?

Comment (rkpagadala) [reply release/pipeline/dags/istio_common_dag.py]: if the file does not exist on remote, copy from local

File: release/pipeline/airflow_scripts.sh (hunk 2 above)
Comment (rkpagadala) [reply release/pipeline/airflow_scripts.sh]: done

File: release/pipeline/airflow_scripts.sh (hunk 1 above)
Comment (rkpagadala) [reply release/pipeline/airflow_scripts.sh]: removed

File: release/pipeline/dags/istio_common_dag.py [hunk 4]
```diff
@@ -138,11 +181,14 @@ def MakeCommonDag(dag_args_func, name,
   )
 
   tasks = dict()
-  cmd_template = getBashSettingsTemplate(extra_param_lst)
+  init_gcb_env_prefix, copy_from_gcb_prefix = getBashSettingsTemplate(extra_param_lst)
 
   def addAirflowBashOperator(cmd_name, task_id, **kwargs):
     cmd_list = []
-    cmd_list.append(cmd_template)
+    if cmd_name == "get_git_commit_cmd":
```
Comment (hklai) [inline release/pipeline/dags/istio_common_dag.py]: this looks hacky

can we just implement a stop to initialize gcb env or bake it in get_git_commit_cmd directly?

File: release/pipeline/dags/istio_common_dag.py [hunk 5]
```diff
@@ -74,46 +74,89 @@ def GetVariableOrDefault(var, default):
...
+"""
+  exp_rel_tools_path_str = 'export CB_GCS_RELEASE_TOOLS_PATH={{ settings.CB_GCS_RELEASE_TOOLS_PATH }}'
+  upload_gcb_env_str = 'gsutil -q cp "${GCB_ENV_FILE}" "gs://${CB_GCS_RELEASE_TOOLS_PATH}/gcb_env.sh"'
+  download_gcb_env_str = 'gsutil -q cp "gs://${CB_GCS_RELEASE_TOOLS_PATH}/gcb_env.sh" "${GCB_ENV_FILE}"'
+  airflow_scripts_init_str = """
+                source "${GCB_ENV_FILE}"
                 # cloning master allows us to use master code to do builds for all releases, if this needs to
                 # be changed because of future incompatible changes
                 # we just need to clone the compatible SHA here and in get_commit.template.json
                 git clone "https://github.com/${CB_GITHUB_ORG}/istio.git" "istio-code" -b "master" --depth 1
```
[the remark this answers is not in the document]
Comment (rkpagadala) [reply release/pipeline/dags/istio_common_dag.py]: leaving this as is for now as discussed

File: release/pipeline/dags/istio_common_dag.py (hunk 4 above)
Comment (rkpagadala) [reply release/pipeline/dags/istio_common_dag.py]: simplified the code a bit using bootstrap files as discussed, and also added an init step

**Comment by rkpagadala** [comment]
@hklai PTAL
